state-chain/pallets/cf-ingress-egress/src/tests.rs

```diff
@@ -1671,3 +1671,140 @@ fn safe_mode_prevents_deposit_channel_creation() {
 		);
 	});
 }
+
+#[test]
+fn do_not_batch_more_transfers_than_the_limit_allows() {
+	new_test_ext().execute_with(|| {
+		MockFetchesTransfersLimitProvider::enable_limits(true);
+
+		const EXCESS_TRANSFERS: usize = 1;
+		let transfer_limits = MockFetchesTransfersLimitProvider::maybe_transfers_limit().unwrap();
+
+		for _ in 1..=transfer_limits + EXCESS_TRANSFERS {
+			assert_ok!(IngressEgress::schedule_egress(ETH_ETH, 1_000, ALICE_ETH_ADDRESS, None));
+		}
+
+		let schedule_egresses = ScheduledEgressFetchOrTransfer::<Test, ()>::get();
+
+		assert_eq!(
+			schedule_egresses.len(),
+			transfer_limits + 1,
+			"Wrong amount of scheduled egresses!"
+		);
+
+		IngressEgress::on_finalize(1);
+
+		let schedule_egresses = ScheduledEgressFetchOrTransfer::<Test, ()>::get();
+
+		assert_eq!(schedule_egresses.len(), EXCESS_TRANSFERS, "Wrong amount of left egresses!");
+
+		IngressEgress::on_finalize(2);
+
+		let schedule_egresses = ScheduledEgressFetchOrTransfer::<Test, ()>::get();
+
+		assert_eq!(schedule_egresses.len(), 0, "Left egresses have been fully processed!");
+
+		MockFetchesTransfersLimitProvider::enable_limits(false);
```

Comment:
Disabling it at the end is not really necessary.

Reply:
It could be since if not the next test that gets executed would run against the enabled limits which might be wrong (depends on the test). If things getting shifted around in the future, this could lead to confusion.

Reply:
That's not really how it works. Every test (everything inside the `#[test] fn x() {..}` runs isolated from each other with fresh state.